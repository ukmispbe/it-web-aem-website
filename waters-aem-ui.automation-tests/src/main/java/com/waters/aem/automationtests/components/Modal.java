package com.waters.aem.automationtests.components;

import com.cognifide.qa.bb.qualifier.PageObjectInterface;


@PageObjectInterface
public interface Modal {

    String getText();

    String getIcon();

}
