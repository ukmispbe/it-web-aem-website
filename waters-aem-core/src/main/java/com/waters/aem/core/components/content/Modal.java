package com.waters.aem.core.components.content;

import com.adobe.cq.export.json.ComponentExporter;
import com.adobe.cq.export.json.ExporterConstants;
import com.citytechinc.cq.component.annotations.Component;
import com.citytechinc.cq.component.annotations.DialogField;
import com.citytechinc.cq.component.annotations.Tab;
import com.citytechinc.cq.component.annotations.widgets.PathField;
import com.citytechinc.cq.component.annotations.widgets.TextField;
import com.icfolson.aem.library.core.components.AbstractComponent;
import com.waters.aem.core.constants.WatersConstants;
import com.waters.aem.core.utils.LinkUtils;
import org.apache.sling.api.SlingHttpServletRequest;
import org.apache.sling.models.annotations.DefaultInjectionStrategy;
import org.apache.sling.models.annotations.Exporter;
import org.apache.sling.models.annotations.Model;

import javax.annotation.Nonnull;
import javax.inject.Inject;

@Component(value = "Modal",
    description = "This is the Modal component for Waters site",
    tabs = @Tab(title = "Properties"))
@Model(adaptables = SlingHttpServletRequest.class,
    adapters = { Modal.class, ComponentExporter.class },
    resourceType = Modal.RESOURCE_TYPE,
    defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
public class Modal extends AbstractComponent implements ComponentExporter {

    public static final String RESOURCE_TYPE = "waters/components/content/modal";

    @DialogField(fieldLabel = "Source Text",
        fieldDescription = "Enter the source text",
        required = true,
        ranking = 2)
    @TextField
    @Inject
    private String sourceText;

    @DialogField(fieldLabel = "Source Icon",
        fieldDescription = "Select source Icon",
        ranking = 2)
    @PathField(rootPath = WatersConstants.DAM_ICON_PATH)
    @Inject
    private String sourceIcon;

    @DialogField(fieldLabel = "Modal Heading Text",
        fieldDescription = "Enter the text modal heading",
        ranking = 3)
    @TextField
    @Inject
    private String modalHeadingText;

    @DialogField(fieldLabel = "Modal Heading Icon",
        fieldDescription = "Select modal heading Icon",
        ranking = 4)
    @PathField(rootPath = WatersConstants.DAM_ICON_PATH)
    @Inject
    private String modalHeadingIcon;

    @DialogField(fieldLabel = "Modal Body Text",
        fieldDescription = "Enter the modal body text",
        ranking = 5)
    @TextField
    @Inject
    private String modalBodyText;

    public String getSourceText() {
        return sourceText;
    }

    public String getSourceIcon() {
        return sourceIcon;
    }

    public String getModalHeadingText() {
        return modalHeadingText;
    }

    public String getModalHeadingIcon() {
        return modalHeadingIcon;
    }

    public String getModalBodyText() {
        return modalBodyText;
    }

    public String getUniqueId() {
        return getId();
    }

    public Boolean isSourceIconSvg() {
        return LinkUtils.isSvg(sourceIcon);
    }

    public Boolean isHeadingIconSvg() {
        return LinkUtils.isSvg(modalHeadingIcon);
    }

    @Nonnull
    @Override
    public String getExportedType() {
        return RESOURCE_TYPE;
    }
}
