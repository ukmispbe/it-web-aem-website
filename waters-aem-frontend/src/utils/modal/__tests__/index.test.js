jest.mock('../modal-portal');

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Modal, { Header } from '../index';

describe('Feature: Modal Component', () => {

    describe('Scenario Rendering Modal', () => {

        describe('When the modal is rendered in a closed state', () => {
            it('Then the snapshot should match', () => {
                const component = renderer.create(
                    <Modal isOpen={false} />
                );

                expect(component).toMatchSnapshot();
            });
        });

        describe('When the modal is rendered in an open state', () => {
            it('Then the snapshot should match', () => {

                const component = renderer.create(
                    <Modal isOpen={true} />
                );

                expect(component).toMatchSnapshot();
            });
        });

        describe('When the modal is rendered w/child components & open state', () => {
            it('Then the snapshot should match', () => {

                let isOpen = true;
                const toggleModal = jest.fn(() => {
                    isOpen = !isOpen;
                })
                    
                const component = renderer.create(
                    <Modal isOpen={true} onClose={toggleModal}>
                        <Header />
                    </Modal>
                );

                expect(component).toMatchSnapshot();
            });
        });

        describe('When the modal is passed a className as a prop', () => {
            it(`Then the wrapper '.cmp-modal-box' should contain that class name`, () => {
                const testClass = 'cmp-test-class';
                const component = shallow(<Modal isOpen={true} className={testClass} />);
                const element = component.find(`.${testClass}`);
                expect(element.length).toEqual(1);
            });
        });

    });

    describe('Scenario Rendering Header Sub-Component', () => {
        describe('When rendering the Header with a title', () => {
            it('Then the snapshot should match', () => {
           
                const props = {
                    title: 'TEST TITLE',
                    onClose: jest.fn(() => {})
                }

                const component = renderer.create(
                    <Header {...props} />
                );
                expect(component).toMatchSnapshot();
            });
        });

        describe('When rendering the Header with a title & Icon', () => {
            it('Then the snapshot should match', () => {
           
                const props = {
                    icon: '/content/dam/waters/en/brand-assets/icons/close.svg',
                    title: 'TEST TITLE',
                    onClose: jest.fn(() => {})
                }

                const component = renderer.create(
                    <Header {...props} />
                );
                expect(component).toMatchSnapshot();
            });
        });
    });
});
