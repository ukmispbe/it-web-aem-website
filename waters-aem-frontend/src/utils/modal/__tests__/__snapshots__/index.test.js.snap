// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Modal Component Scenario Rendering Header Sub-Component When rendering the Header with a title & Icon Then the snapshot should match 1`] = `
<div
  className="cmp-modal-box "
  onClick={[Function]}
>
  <div
    className="cmp-modal"
  >
    <div
      className="cmp-modal__header header-with-title "
      data-locator="test-title"
    >
      <div
        className="cmp-modal__title"
      >
        <div
          className="cmp-modal__title-icon"
          data-locator="header-icon"
        >
          <div />
        </div>
        <div
          aria-label="TEST TITLE"
          className="cmp-modal__title-text"
          data-locator="test-title"
          role="heading"
        >
          TEST TITLE
        </div>
      </div>
      <button
        className="cmp-modal__close-icon"
        onClick={[MockFunction]}
      >
        <div
          aria-hidden="true"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Feature: Modal Component Scenario Rendering Header Sub-Component When rendering the Header with a title Then the snapshot should match 1`] = `
<div
  className="cmp-modal-box "
  onClick={[Function]}
>
  <div
    className="cmp-modal"
  >
    <div
      className="cmp-modal__header header-with-title "
      data-locator="test-title"
    >
      <div
        className="cmp-modal__title"
      >
        <div
          aria-label="TEST TITLE"
          className="cmp-modal__title-text"
          data-locator="test-title"
          role="heading"
        >
          TEST TITLE
        </div>
      </div>
      <button
        className="cmp-modal__close-icon"
        onClick={[MockFunction]}
      >
        <div
          aria-hidden="true"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Feature: Modal Component Scenario Rendering Modal When the modal is rendered in a closed state Then the snapshot should match 1`] = `null`;

exports[`Feature: Modal Component Scenario Rendering Modal When the modal is rendered in an open state Then the snapshot should match 1`] = `
<div
  className="cmp-modal-box "
  onClick={[Function]}
>
  <div
    className="cmp-modal"
  />
</div>
`;

exports[`Feature: Modal Component Scenario Rendering Modal When the modal is rendered w/child components & open state Then the snapshot should match 1`] = `
<div
  className="cmp-modal-box "
  onClick={[Function]}
>
  <div
    className="cmp-modal"
  >
    <div
      className="cmp-modal__header  "
      data-locator=""
    >
      <button
        className="cmp-modal__close-icon"
        onClick={[MockFunction]}
      >
        <div
          aria-hidden="true"
        />
      </button>
    </div>
  </div>
</div>
`;
